Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/CopyPass.h

```diff
@@ -101,6 +109,37 @@ namespace AZ
             int m_currentBufferIndex = 0;
             AZStd::array<Ptr<RHI::Fence>, MaxFrames> m_device1SignalFence;
             AZStd::array<Ptr<RHI::Fence>, MaxFrames> m_device2WaitFence;
+            // RPI::Pass overrides...
+            TimestampResult GetTimestampResultInternal() const override;
+            PipelineStatisticsResult GetPipelineStatisticsResultInternal() const override;
+
+            // Helper function to get the query by the scope index and query type
+            RHI::Ptr<Query> GetQuery(ScopeQueryType queryType, int scopeIndex);
+
+            // Executes a lambda depending on the passed ScopeQuery types
+            template<typename Func>
+            void ExecuteOnTimestampQuery(Func&& func, int scopeIndex);
+            template<typename Func>
+            void ExecuteOnPipelineStatisticsQuery(Func&& func, int scopeIndex);
+
+            // Begin recording commands for the ScopeQueries
+            void BeginScopeQuery(const RHI::FrameGraphExecuteContext& context, int scopeIndex);
```

Comment:
To be honest the word scopeIndex was confusing. To me it implies an index related to RHI::Scope and not a specific device.